curvesim/tools/bonding_curve.py

```diff
@@ -49,49 +55,104 @@ def bonding_curve(pool, *, truncate=0.0005, resolution=1000, plot=False):
+        combos = list(combinations(range(pool.n), 2))
+
+    xp: List[int] = pool._xp()  # pylint: disable=protected-access
+
+    if isinstance(pool, STABLESWAP):  # type: ignore[misc, arg-type]
+        D: int = pool.D()  # type: ignore[assignment]
+        if truncate is None:
+            # pylint: disable=pointless-string-statement
+            """
+            This default value works for Stableswap, but will break Cryptoswap.
+            At this value, the graph usually cuts off cleanly around the points where
+            the Stableswap pool would depeg, as one stablecoin balance has reached
+            almost 100% of pool assets.
+            """
+            truncate = 0.0005
+    elif isinstance(pool, CRYPTOSWAP):  # type: ignore[misc, arg-type]
+        D = pool.D  # Don't recalcuate D - it will rebalance the bonding curve(s)
+        if truncate is None:
+            # pylint: disable=pointless-string-statement
+            """
+            A 1.0 value for Cryptoswap extends the graph to a point at which the pool
+            would incur massive losses if it were to rebalance. The further away from
+            (1 / pool.n) truncate is, the more imbalanced the pool is at the end of the
+            graph before rebalancing, if profit is high enough to rebalance.
+            """
+            truncate = 1.0
+    else:
+        raise TypeError(f"Bonding curve calculation not supported for {type(pool)}")
 
-    pair_to_curve = {}
+    pair_to_curve: Dict[Tuple[int, int], List[Tuple[float, float]]] = {}
+    current_points: Dict[Tuple[int, int], Tuple[float, float]] = {}
     for (i, j) in combos:
-        truncated_D = int(D * truncate)
-        x_max = pool.get_y(j, i, truncated_D, xp)
-        xs = linspace(truncated_D, x_max, resolution).round()
+        truncated_D: int = int(D * truncate)
+        x_limit: int = pool.get_y(j, i, truncated_D, xp)
+        xs: List[int] = list(linspace(truncated_D, x_limit, resolution).round())
 
-        curve = []
+        curve: List[Tuple[float, float]] = []
         for x in xs:
-            y = pool.get_y(i, j, int(x), xp)
-            curve.append((x, y))
-        curve = [(x / D_UNIT, y / D_UNIT) for x, y in curve]
+            x_float: float = x / D_UNIT
+            y_float: float = pool.get_y(i, j, int(x), xp) / D_UNIT
+
+            if isinstance(pool, CRYPTOSWAP):  # type: ignore[misc, arg-type]
+                if i > 0:
+                    # type: ignore[union-attr]
+                    x_float = x_float * D_UNIT / pool.price_scale[i - 1]
+
+                if j > 0:
+                    # type: ignore[union-attr]
+                    y_float = y_float * D_UNIT / pool.price_scale[j - 1]
+
+            curve.append((x_float, y_float))
+
         pair_to_curve[(i, j)] = curve
 
+        current_x: float = xp[i] / D_UNIT
+        current_y: float = xp[j] / D_UNIT
+
+        if isinstance(pool, CRYPTOSWAP):  # type: ignore[misc, arg-type]
```

Comment:
would be better to use a function that adjusts the amount properly depending on the pool type; you're basically writing the same thing 4 times already and it will get unmanageable if there are more types.